### PR TITLE
Move to GitHub action for releases

### PR DESCRIPTION
**Pull request by jacobbednarz**
Introduces GitHub Actions for deploying the changes to wordpress.org instead of 
our existing manual process.

In it's current form, the TL;DR of the manual process looks like this ([full details](https://github.com/cloudflare/Cloudflare-WordPress/tree/master/scripts/deployment)).

- Update [backend](https://github.com/cloudflare/cloudflare-plugin-backend)/[frontend](https://github.com/cloudflare/cloudflare-plugin-frontend) repositories with whatever changes are required. 
- Update README with version bumps and CHANGELOG entries.
- Draft a new GitHub release (without artifacts).
- Clean up the repository and build the intended structure for SVN.
- Commit, tag and push the SVN changes to wordpress.org
- Update the GitHub.

As the linked documentation shows, this is a pretty complex workflow with 
plenty of places that people could fall into the abyss without a great recovery 
plan.

To assist with automation and consistency, I've ported majority of the steps 
into a GitHub Action that is triggered on release. The updated workflow is now:

- Update the backend/frontend repositories.
- Update README with version bumps and CHANGELOG entries.
- Create a new GitHub release via the UI.
- Profit ✨

The SVN interactions, repository packaging and directory structure munging is 
now handled behind the scenes. The `release` action will handle generating the
correct structure, committing it and uploading the generated artifact back to 
the GitHub Action.

### CODE REVIEW

File: .distignore [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/.git
```
Comment (jacobbednarz) [inline .distignore:1]: This file supersedes https://github.com/cloudflare/Cloudflare-WordPress/blob/master/scripts/deployment/trim_code_for_release.sh and once this is merged, we can redo/cleanup the documentation to reflect the new steps.

File: scripts/deployment/trim_code_for_release.sh [hunk 2]
```diff
@@ -1,47 +0,0 @@
-#!/bin/bash
```
Comment (jacobbednarz) [inline scripts/deployment/trim_code_for_release.sh:1]: this is superseded by `.distignore`

**Review by patryk — APPROVED** [review]

**Comment by sejoker** [comment]
Thank's for working on this, I'm looking forward to test it out, once merged.

File: docs/release-new-version.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Releasing New Version Of Cloudflare WordPress Plugin
```
Comment (manatarms) [inline docs/release-new-version.md:1]: This is so much better. Thank you

**Review by manatarms — APPROVED** [review]

**Review by mgirouard — APPROVED** [review]

**Pull request merged**